src/traces/violin/index.js

```diff
@@ -0,0 +1,40 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+module.exports = {
+    attributes: require('./attributes'),
+    layoutAttributes: require('./layout_attributes'),
+    supplyDefaults: require('./defaults'),
+    supplyLayoutDefaults: require('./layout_defaults'),
+    calc: require('./calc'),
+    setPositions: require('../box/set_positions'),
+    plot: require('./plot'),
+    style: require('./style'),
+    hoverPoints: require('../box/hover'),
+    selectPoints: require('../box/select'),
+
+    moduleType: 'trace',
+    name: 'violin',
+    basePlotModule: require('../../plots/cartesian'),
+    // TODO
+    // - should maybe rename 'box' category to something more general
+    categories: ['cartesian', 'symbols', 'oriented', 'box', 'showLegend'],
```

Comment:
this `box` category is currently used:

![image](https://user-images.githubusercontent.com/6675409/31955870-ba61d104-b8b7-11e7-9eca-dac5615d5e92.png)

Reply:
... which for the most part are common places for `violin` traces too.

Reply:
Replaced by `box-violin` in https://github.com/plotly/plotly.js/pull/2116/commits/ea43b251428047d63afc2a332d308796700a11ac